src/librbd/Operations.cc

```diff
@@ -779,7 +779,9 @@ int Operations<I>::snap_rollback(const cls::rbd::SnapshotNamespace& snap_namespa
   }
 
   r = prepare_image_update();
-  if (r < 0) {
+  if (r == -EAGAIN) {
```

Comment:
Fair enough -- I forgot that I wrote rollback as a best-effort where it doesn't acquire the lock if someone else already has it (since it doesn't make any sense to be rolling back image-state on an actively in-use image).